core/services/ocr2/delegate.go

```diff
@@ -820,6 +825,159 @@ func (d *Delegate) newServicesVaultPlugin(
 	}
 	srvs = append(srvs, job.NewServiceAdapter(oracle))
 
+	// lggr2 := logger.Sugared(lggr.With("plugin", "dkg", "contractID", "0x0165878A594ca255338adfa4d48449f69242Eb8F"))
+	// ocrLogger2 := ocrcommon.NewOCRWrapper(lggr2, d.cfg.OCR2().TraceLogging(), func(ctx context.Context, msg string) {
+	// 	lggr2.ErrorIf(d.jobORM.RecordError(ctx, jb.ID, msg), "unable to record error")
+	// })
+	// srvs = append(srvs, ocrLogger2)
+
+	// fmt.Printf("DOING SECOND OCR ORACLE %s %s %s\n%x %x\n", spec.RelayConfig, spec.TransmitterID.String, spec.PluginConfig, onchainKeyringAdapter.PublicKey(), kb.OffchainPublicKey())
+	// provider2, err := relayer.NewPluginProvider(ctx, types.RelayArgs{
+	// 	ExternalJobID: jb.ExternalJobID,
+	// 	JobID:         jb.ID,
+	// 	OracleSpecID:  spec.ID,
+	// 	ContractID:    "0x0165878A594ca255338adfa4d48449f69242Eb8F",
+	// 	New:           d.isNewlyCreatedJob,
+	// 	RelayConfig:   spec.RelayConfig.Bytes(),
+	// 	ProviderType:  string(types.OCR3Capability),
+	// }, types.PluginArgs{
+	// 	TransmitterID: spec.TransmitterID.String,
+	// 	PluginConfig:  []byte{},
+	// })
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// srvs = append(srvs, provider2)
+
+	// fullPathDKG := filepath.Join(d.cfg.OCR2().KeyValueStoreRootDir(), jb.ExternalJobID.String(), "_dkg")
+	// err = utils.EnsureDirAndMaxPerms(fullPathDKG, os.FileMode(0700))
+	// if err != nil {
+	// 	return nil, fmt.Errorf("failed to create key value store directory: %w", err)
+	// }
+	// oracleEndpoint2 := d.monitoringEndpointGen.GenMonitoringEndpoint(
+	// 	rid.Network,
+	// 	rid.ChainID,
+	// 	"0x0165878A594ca255338adfa4d48449f69242Eb8F",
+	// 	synchronization.TelemetryType("dkg-plugin"),
+	// )
+
+	// oracleArgs2 := oracleargs.OCR3_1OracleArgsForSanMarinoDKG(
+	// 	d.peerWrapper.Peer3_1,
+	// 	bootstrapPeers,
+	// 	provider2.ContractConfigTracker(),
+	// 	ocrDB,
+	// 	kvdb.NewBadgerKeyValueDatabaseFactory(fullPathDKG),
+	// 	lc,
+	// 	ocrLogger2,
+	// 	prometheus.WrapRegistererWith(map[string]string{"job_name": "dkg_job"}, prometheus.DefaultRegisterer),
+	// 	oracleEndpoint2,
+	// 	provider2.OffchainConfigDigester(),
+	// 	kb,
+	// 	dkgRecipientKey,
+	// 	vault.NewVaultORM(d.ds),
+	// 	common.HexToAddress("0x0165878A594ca255338adfa4d48449f69242Eb8F"),
+	// )
+	// oracle2, err := libocr2.NewOracle(oracleArgs2)
+	// if err != nil {
+	// 	return nil, err
+	// }
+	// srvs = append(srvs, job.NewServiceAdapter(oracle2))
+
+	return srvs, nil
+}
+
+func (d *Delegate) newServicesDKGPlugin(
+	ctx context.Context,
+	lggr logger.SugaredLogger,
+	jb job.Job,
+	bootstrapPeers []commontypes.BootstrapperLocator,
+	kb ocr2key.KeyBundle,
+	ocrDB *db,
+	lc ocrtypes.LocalConfig,
+	capabilitiesRegistry core.CapabilitiesRegistry,
+	wrapper *gatewayconnector.ServiceWrapper,
+	syncer syncerV2.WorkflowRegistrySyncer,
+) (srvs []job.ServiceCtx, err error) {
+	spec := jb.OCR2OracleSpec
+
+	dkgRecipientKeys, err := d.dkgRecipientKs.GetAll()
```

Comment:
There should be a helper you can use to get this key btw, see https://github.com/smartcontractkit/chainlink/pull/19301/files#diff-ee42cd8d3c2284e2aa914b4f7797b308ef4bf9f3e73a5692da84685f3bf9f542R808